app/views/course/courses/show.html.slim

```diff
@@ -15,13 +15,15 @@
                 collection: current_course.managers.includes(:user).map(&:user)
 
   - if current_course.user?(current_user) || can?(:manage, current_course)
-    h2 = t('.announcements')
-    div.message-holder
-      = render partial: current_course.announcements.currently_active, spacer_template: 'announcements/announcement_spacer'
+    - if !current_course.announcements.currently_active.empty?
```

Comment:
Yes. Make the queries before you render.

On Thu, Sep 8, 2016 at 10:49 PM, ALLEN WANG <notifications@github.com>
wrote:

> In app/views/course/courses/show.html.slim
> <https://github.com/Coursemology/coursemology2/pull/1465#discussion_r78021469>
> :
>
> > @@ -15,13 +15,15 @@
> >                  collection: current_course.managers.includes(:user).map(&:user)
> >
> >    - if current_course.user?(current_user) || can?(:manage, current_course)
> > -    h2 = t('.announcements')
> > -    div.message-holder
> > -      = render partial: current_course.announcements.currently_active, spacer_template: 'announcements/announcement_spacer'
> > +    - if !current_course.announcements.currently_active.empty?
>
> Cache the query result to a variable ?
>
> Otherwise there are 2 queries generated if there're any announcements.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Coursemology/coursemology2/pull/1465/files/cd7124ce373e82b9610335e8cc7a0db1c52dd413#r78021469>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPMdQd_1YPNHksY8Q0n_5ObO3HDYp_Gks5qoCB0gaJpZM4J3z1S>
> .
>